lib/base/scriptutils.cpp

```diff
@@ -520,7 +520,7 @@ String ScriptUtils::MsiGetComponentPathShim(const String& component)
 
 Array::Ptr ScriptUtils::TrackParents(const Object::Ptr& child)
 {
-	return Array::FromVector(DependencyGraph::GetParents(child));
+	return Array::FromVector(DependencyGraph::GetChildren(child));
```

Comment:
The method is called `TrackParents` and not children, how is this supposed to work now?

```bash
<4> => get_object(Dependency, "brother!mother")
{
	__name = "brother!mother"
	active = true
	child_host_name = "brother"
	child_service_name = ""
	disable_checks = false
	disable_notifications = true
        ....
<5> => track_parents(get_object(Dependency, "brother!mother"))
[ ]
<8> => track_parents(get_host("mother"))
[ {
	__name = "brother!mother"
	active = true
	child_host_name = "brother"
	child_service_name = ""
	disable_checks = false
	disable_notifications = true
...
```

Reply:
Then it has never worked, because I literally just renamed methods and method args.

Reply:
If we want to also fix this, it would be a breaking change. (I'd even drop the method, so that its users notice easily.) But this is something for v2.15, i.e another PR. (And no, I won't open another PR right now, don't worry.)

Reply:
If @julianbrost wants to merge this while that DSL function is still broken (was broken before too), then it would probably work as expected with #10000<sub>like for the first time since it was introduced :)</sub>.

Reply:
And the best discovery tool for real-world usages is removal.😈

Reply:
But yes, your fact checking is correct:

```
$ git grep -nFie track_parents
lib/base/scriptutils.cpp:52:REGISTER_SAFE_FUNCTION(System, track_parents, &ScriptUtils::TrackParents, "child");
$ git grep -nFie track\\_parents
$ git grep -nFie track\\\\_parents
$
```

Same at Google btw.: https://letmegooglethat.com/?q=%22icinga%22+2+%22track_parents%22

Reply:
Yay 🎉! It works now with d28dfeceecf8f21fc60037d235718549d1b4867b!

```bash
<11> => track_parents(get_object(Dependency, "brother!mother"))
[ {
	__name = "mother"
	acknowledgement = 0.000000
	acknowledgement_expiry = 0.000000
	acknowledgement_last_change = 0.000000
	action_url = ""
	active = true
	address = ""
	address6 = ""
	check_attempt = 1.000000
	check_command = "dummy"
	check_interval = 60.000000
	check_period = ""
	check_timeout = null
	command_endpoint = ""
	display_name = "mother"
	downtime_depth = 0.000000
	enable_active_checks = true
	enable_event_handler = true
	enable_flapping = false
	enable_notifications = true
	enable_passive_checks = true
....
```